src/main/webapp/app/exam/manage/test-runs/test-run-management.component.ts

```diff
@@ -0,0 +1,134 @@
+        private jhiAlertService: JhiAlertService,
+        private examManagementService: ExamManagementService,
+        private accountService: AccountService,
+        private sortService: SortService,
+        private modalService: NgbModal,
+    ) {
+        this.predicate = 'id';
+        this.ascending = true;
+    }
+
+    ngOnInit(): void {
+        this.examManagementService.find(Number(this.route.snapshot.paramMap.get('courseId')), Number(this.route.snapshot.paramMap.get('examId')), false, true).subscribe(
+            (response: HttpResponse<Exam>) => {
+                this.exam = response.body!;
+                this.isExamStarted = this.exam.started;
+                this.course = this.exam.course;
+                this.course.isAtLeastInstructor = this.accountService.isAtLeastInstructorInCourse(this.course);
+                this.examManagementService.findAllTestRunsForExam(this.course.id, this.exam.id).subscribe(
+                    (res) => {
+                        this.testRuns = res.body!;
+                    },
+                    (err) => this.onError(err),
+                );
+            },
+            (error) => this.onError(error),
+        );
+        this.accountService.fetch().subscribe((res) => {
+            if (res.body != null) {
+                this.instructor = res.body;
+            }
+        });
+    }
+
+    /**
+     * Open modal to configure a new test run
+     */
+    openCreateTestRunModal() {
+        const modalRef: NgbModalRef = this.modalService.open(CreateTestRunModalComponent as Component, { size: 'lg', backdrop: 'static' });
+        modalRef.componentInstance.exam = this.exam;
+        modalRef.result
+            .then((testRunConfiguration: StudentExam) => {
+                this.examManagementService.createTestRun(this.course.id, this.exam.id, testRunConfiguration).subscribe(
+                    (res) => {
+                        if (res.body != null) {
+                            this.testRuns.push(res.body!);
+                        }
+                    },
+                    (error) => {
+                        this.onError(error);
+                    },
+                );
+            })
+            .catch(() => {});
+    }
+
+    /**
+     * Delete the test run with the given id.
+     * @param testRunId {number}
+     */
+    deleteTestRun(testRunId: number) {
+        this.examManagementService.deleteTestRun(this.course.id, this.exam.id, testRunId).subscribe(
+            () => {
+                this.testRuns = this.testRuns!.filter((testRun) => testRun.id !== testRunId);
+                this.dialogErrorSource.next('');
+            },
+            (error) => this.dialogErrorSource.next(error.message),
+        );
+    }
+
+    /**
+     * Track the items on the testruns Table
+     * @param index {number}
+     * @param item {StudentExam}
+     */
+    trackId(index: number, item: StudentExam) {
+        return item.id;
+    }
```

Comment:
Could you just use `item.id` where ever this function is called?

Reply:
This is required for the angular change detection, to track the items by id